src/arrayMethodJoin.js

```diff
@@ -4,8 +4,39 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function (separator = ',') {
+    let joinedString = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this.length === 0) {
+      return [];
+    }
+
+    if (this.length === 1) {
+      return '' + this[0];
+    }
+
+    if (!this[1] && this[0]) {
```

Comment:
a check for !this[1] && this[0] that seems unnecessary and unclear in its intent